src/main/java/mediator/Mediator.java

```diff
@@ -30,409 +32,412 @@
+                                                Double transmittingPower) {
+        ManagedRadioUnit ru = getRadioUnit(ip);
+        if (ru == null) {
+            System.out.println("Unable to create carrier on Radio Unit: Radio Unit not found!");
+            return;
+        }
+        ru.setupCarrier(createCarrier(rfPorts, carrierFrequencies, transmittingPower, ru.getRatType()));
+    }
+
+    /**
+     * Return a radio unit specified by a given IP address.
+     *
+     * @param ip The IP of the RU, as a String.
+     * @return The radio unit associated with that specific ip.
+     */
+    private ManagedRadioUnit getRadioUnit(String ip) {
+        return radioUnitRegistry.getByIpAddress(ip);
+    }
+
+    /**
+     * Create a carrier that will then be registered with the system. The carrier will have its type
+     * determined based on the RAT type of the radio unit the carrier is being added to.
+     *
+     * @param rfPorts            A list of RF ports that will be used for this carrier.
+     * @param carrierFrequencies The frequency band that will be used with this carrier.
+     * @param transmittingPower  The transmitting power for this carrier.
+     * @param ratType            The RAT type of the radio unit that this carrier will be associated with.
+     * @return The created Carrier based on the RAT type of the radio unit it will be associated with.
+     */
+    private synchronized Carrier createCarrier(List<RfPort> rfPorts, FrequencyBand carrierFrequencies, Double transmittingPower, RatType ratType) {
+        switch (ratType) {
+            case WCDMA: {
+                try {
+                    return carrierManagement.createWcdmaCarrier(rfPorts, carrierFrequencies, transmittingPower);
+                } catch (ArrayIndexOutOfBoundsException ex) {
+                    return null;
+                }
+            }
+            case LTE: {
+                try {
+                    return carrierManagement.createLteCarrier(rfPorts, carrierFrequencies, transmittingPower);
+                } catch (ArrayIndexOutOfBoundsException ex) {
+                    return null;
+                }
+            }
+            default: {
+                System.out.print("[ERROR] Invalid RAT type detected, failed to create a carrier.");
+                return null;
+            }
+        }
+    }
+
+    /**
+     * List any registered radio units that contain the parameter being passed.
+     *
+     * @param param The parameter that will be used to query for radio units. Examples
+     *              include a RatType, a RadioUnitState, a FrequencyBand, or a String name.
+     */
+    private void listRuByParam(Object param) {
+        if (param instanceof RatType) {
+            List<ManagedRadioUnit> radioUnits = radioUnitRegistry.getByRatType((RatType) param);
+            if (radioUnits.size() == 0) {
+                System.out.printf("No Radio Units with the RAT type \"%s\" exist in the system.\n", param);

```

Comment:
I think this is a good addition for the list methods. Thanks for keeping them consistent.